File: WebApp/Models/Department.cs
namespace WebApp.Models
{
    public class Department : BaseEntity
    {
        public override long Id { get; set; }
        public override string Name { get; set; }
        public string Boss { get; set; }
        public long CompanyId { get; set; }
        public Company CompanyEntity;
        public Department() { }// { company = new Company(); }
        public Department(long companyId) { Id = 0; Name = ""; Boss = ""; CompanyId = companyId; }
        public Department(string name, string boss, long companyId) { Id = 0; Name = name; Boss = boss; CompanyId = companyId; }
        public override string ToString() { return "{Id = " + Id.ToString() + ", Name = " + Name + ", Boss = " + Boss + ", Company = " + CompanyEntity?.ToString() + "}"; }
    }

}

File: WebApp/ClientApp/src/app/department/department.ts
import { Company } from '../company/company';

export class Department {
  public companyEntity: Company;

  constructor(
    public id: number,
    public name: string,
    public boss: string,
    public companyId: number
  ) { this.companyEntity = new Company(0, "", ""); }
}
